plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,507 @@
+        type: str
+        choices: [ 'nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1', 'ed25519' ]
+    key_usage:
+        description:
+            - Specifies usage(s) for key.
+            - V(cert) is given to all primary keys regardess, however can be used to only give V(vert) usage to a key.
+            - If not usage is specified, all of valid usages for the given key type are assigned.
+            - O(key_usage=encr) is only supported is O(key_type=RSA).
+        type: list
+        elements: str
+        default: []
+        choices: [ 'encr', 'sign', 'auth', 'cert' ]
+    subkeys:
+        description:
+            - List of subkeys with their own respective key types, lengths, curves, and usages.
+        type: list
+        elements: dict
+        default: []
+        suboptions:
+            subkey_type:
+                description:
+                 - Similar to O(key_type).
+                 - Also supports ECDH and ELG keys.
+                type: str
+                choices: [ 'RSA', 'DSA', 'ECDSA', 'EDDSA', 'ECDH', 'ELG' ]
+            subkey_length:
+                description:
+                    - Similar to O(key_length).
+                    - For ELG subkeys, the minimum length is V(1024) bits, the maximum length is V(4096) bits, and the default length is V(3072) bits.
+                type: int
+            subkey_curve:
+                description:
+                    - Similar to O(key_curve).
+                    - V(cv25519) is supported if subkey_type is V(ECDH).
+                    - This is required if subkey_type is V(ECDSA), V(EDDSA), or V(ECDH) and it is ignored if subkey_type is V(RSA), V(DSA), or V(ELG).
+                type: str
+                choices: ['nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1', 'ed25519', 'cv25519']
+            subkey_usage:
+                description:
+                    - Similar to O(key_usage).
+                    - V(encr) is supported if subkey_type is V(RSA), V(ECDH), or V(ELG).
+                    - If subkey_type is V(ECDH) or V(ELG), only V(encr) is supported.
+                type: list
+                elements: str
+                default: []
+                choices: [ 'encr', 'sign', 'auth' ]
+    expire_date:
+        description:
+            - Sets the expire date for the key.
+            - If O(expire_date=0), the key never expires.
+            - If O(expire_date=<n>), the key expires in V(n) days.
+            - If O(expire_date=<n>w), the key expires in V(n) weeks.
+            - If O(expire_date=<n>m), the key expires in V(n) months.
+            - If O(expire_date=<n>y), the key expires in V(n) years.
+            - If left unspecified, any created GPG keys never expire.
+        type: str
+    name:
+        description:
+            - Specifies a name for the key's user id.
```

Comment:
```suggestion
            - Specifies a name for the key's user ID.
```
(https://docs.ansible.com/ansible/devel/dev_guide/style_guide/spelling_word_choice.html#acronyms)